src/Raven.Server/Smuggler/Documents/DatabaseSmuggler.cs

```diff
@@ -250,6 +251,22 @@ private void ProcessType(DatabaseItemType type, SmugglerResult result, BuildVers
             _onProgress.Invoke(result.Progress);
         }
 
+        private void TryProcessDocumentsWithDuplicateCollection(SmugglerResult result, BuildVersionType buildType)
+        {
+            if (_destination is DatabaseDestination databaseDestination && databaseDestination._duplicateDocsHandler.DocumentsWithDuplicateCollection.Count > 0)
```

Comment:
If you need to do the type check here then it is modelled incorrectly.

Reply:
introduce in destination interface method: `GetDocumentsWithDuplicateCollection` and introduce a parameter to actions.Documents for handling the duplicates instead of changing some internal flag.